engine/common/api_helpers/custom_rate_scoped_throttler.py

```diff
@@ -1,55 +1,30 @@
-from rest_framework.throttling import SimpleRateThrottle
+from django.conf import settings
+from ratelimit.utils import _split_rate
+from rest_framework.throttling import UserRateThrottle
 
 
-class CustomRateScopedThrottler(SimpleRateThrottle):
```

Comment:
Refactored this class for public api

Reply:
After taking a look at PR, I figured out what was the idea of CustomRateScopedThrottler: Because it had an abstract get_scope method it was not possible to accidentally create new throttler with same scope, by forgetting specify self.scope in a new class.